torch/csrc/autograd/record_function.cpp

```diff
@@ -230,16 +222,10 @@ void RecordFunction::end() {
   if (initialized_) {
     runEndCallbacks();
 
-    // In the case that RecordFunction::end is called from a different thread,
-    // thread_local_func will not be this, so assert that we have overridden the
-    // thread id (by ensuring it is nonzero) and thread_local_func is null.
     TORCH_INTERNAL_ASSERT(
-        (thread_local_func_ == this) ||
-            (thread_local_func_ == nullptr && threadId_ != 0),
+        (thread_local_func_ == this),
         name_,
-        ": must be top of stack. If you are calling RecordFunction::end in a"
-        "separate thread, call RecordFunction::setThreadId() in the creating"
-        "thread.");
+        ": must be top of stack.");
     thread_local_func_ = parent_;
```

Comment:
This is just reverting the changes to RecordFunction from previous PRs right?

Reply:
Yes, since we have a RecordFnAsync now we can undo the changes to RecordFn and have them in the async one.